tests/integration/run_tests.py

```diff
@@ -317,6 +362,69 @@ async def test_getting_all_objects_from_schema_works():
         await asyncio.gather(*[delete_object(sess, "study", accession_id) for accession_id in accession_ids])
 
 
+async def test_crud_folders_works():
+    """Test folders REST api POST, GET, PATCH and DELETE reqs."""
+    async with aiohttp.ClientSession() as sess:
+        # Create new folder and check it creation succeeded
+        data = {"name": "test", "description": "test folder"}
+        folder_id = await post_folder(sess, data)
+        async with sess.get(f"{folders_url}/{folder_id}") as resp:
+            LOG.debug(f"Checking that folder {folder_id} was created")
+            assert resp.status == 200, "HTTP Status code error"
+
+        # Add object to session and create a patch to add object to folder
+        accession_id = await post_object(sess, "sample", "SRS001433.xml")
+        patch = [{"op": "add", "path": "/metadataObjects/0", "value": {accession_id, "sample"}}]
+        folder_id = await patch_folder(sess, folder_id, patch)
+        async with sess.get(f"{folders_url}/{folder_id}") as resp:
+            LOG.debug(f"Checking that folder {folder_id} was patched")
+            folder = {
+                "folderId": folder_id,
+                "name": "test",
+                "description": "test folder",
+                "published": False,
+                "metadataObjects": [{accession_id, "sample"}],
+            }
+            assert resp.json == folder, "Response data error"
```

Comment:
i would assert individual values to be sure. See https://github.com/CSCfi/metadata-submitter/blob/master/tests/integration/run_tests.py#L202-L219

Reply:
Good point